crates/rooch-rpc-server/src/service/rpc_service.rs

```diff
@@ -358,13 +358,23 @@ impl RpcService {
         show_display: bool,
         state_type: ObjectStateType,
     ) -> Result<Vec<IndexerObjectStateView>> {
+        const MAX_OBJECT_IDS: usize = 100;
+
         let indexer_ids = match filter {
             // Compatible with object_ids query after split object_states
             // Do not query the indexer, directly return the states query results.
-            ObjectStateFilter::ObjectId(object_ids) => object_ids
-                .into_iter()
-                .map(|v| (v, IndexerStateID::default()))
-                .collect(),
+            ObjectStateFilter::ObjectId(object_ids) => {
```

Comment:
There are also scenarios where queryUTXOs, queryInscriptions, repairIndexer, getStates, and listStates API all have batch query ObjectIDs, which can be considered for fixup in the next PR